src/App.tsx

```diff
@@ -1,24 +1,178 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import {
+  FC,
+  useState,
+  useEffect,
+  useMemo,
+  useCallback,
+} from 'react';
+import { TodoService } from './api/todos';
 
-const USER_ID = 0;
+import { Todo } from './Types/Todos';
+import { TodoStatus } from './Types/TodoStatus';
+import { filterTodos } from './Utils/Helpers';
 
-export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+import { AddingForm } from './components/AddingForm/AddingForm';
+import { TodoList } from './components/TodoList/TodoList';
+import { TodoFilter } from './components/TodoFilter/TodoFilter';
+import { NotificationError }
+  from './components/NotificationError/NotificationError';
+import { Errors } from './Types/Errors';
+
+const USER_ID = 6922;
+
+export const App: FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [todoStatus, setTodoStatus] = useState<TodoStatus>(TodoStatus.ALL);
+  const [loadingTodosIds, setLoadingTodosIds] = useState<number[]>([]);
+  const [isInputDisabled, setIsInputDisabled] = useState<boolean>(false);
+  const [error, setError] = useState<Errors>(Errors.NONE);
+
+  const loadTodos = async () => {
+    try {
+      const loadedTodos = await TodoService.getByUserId(USER_ID);
+
+      setTodos(loadedTodos);
+    } catch (err) {
+      setError(Errors.LOAD);
+    }
+  };
+
+  useEffect(() => {
+    loadTodos();
+  }, []);
+
+  const visibleTodos = useMemo(() => {
+    return filterTodos(todos, todoStatus);
+  }, [todos, todoStatus]);
+
+  const numberOfRemainingTodos = useMemo(() => {
+    return filterTodos(todos, TodoStatus.ACTIVE).length;
+  }, [todos]);
+
+  const addTodo = useCallback(async (todoData: Omit<Todo, 'id'>) => {
+    setIsInputDisabled(true);
+    const temporaryTodo = {
+      id: 0,
+      ...todoData,
+    };
+
+    try {
+      setTempTodo(temporaryTodo);
+      setLoadingTodosIds(currIds => [...currIds, temporaryTodo.id]);
+      const addedTodo = await TodoService.create(temporaryTodo);
+
+      setTodos(prevTodos => [...prevTodos, addedTodo]);
+    } catch (err) {
+      setError(Errors.ADD);
+    } finally {
+      setTempTodo(null);
+    }
+
+    setIsInputDisabled(false);
+  }, []);
```

Comment:
None of child components is optimized, please, read when necessary to use useCallback hook, and make sure to delete it everywhere, where it's not necessary